### PR TITLE
add sync error info and sync error type

### PR DESCRIPTION
**Pull request by sun20121818**

### CODE REVIEW

File: src/message-poller.cpp [hunk 1]
```diff
@@ -187,6 +189,12 @@ void MessagePoller::processNotification(const SyncNotification& notification)
         case SYNC_ERROR_ID_FOLDER_PERM_DENIED:
             msg = tr("Update to file %1 is denied by folder permission setting.").arg(path);
             break;
+        case SYNC_ERROR_ID_PERM_NOT_SYNCABLE:
+            msg = tr("No permission to sync this folder.").arg(path);
+            break;
+        case SYNC_ERROR_ID_UPDATE_TO_READ_ONLY_REPO:
+            msg = tr("Updates in read-only library will not be uploaded.").arg(path);
```
Comment (lins05) [inline src/message-poller.cpp]: 字符串里面没有 `%1`, 后面为什么需要 `.arg(path` 呢?

Comment (sun20121818) [reply src/message-poller.cpp]: ok

File: src/message-poller.cpp [hunk 2]
```diff
@@ -187,6 +189,12 @@ void MessagePoller::processNotification(const SyncNotification& notification)
         case SYNC_ERROR_ID_FOLDER_PERM_DENIED:
             msg = tr("Update to file %1 is denied by folder permission setting.").arg(path);
             break;
+        case SYNC_ERROR_ID_PERM_NOT_SYNCABLE:
+            msg = tr("No permission to sync this folder %s.").arg(path);
```
Comment (lins05) [inline src/message-poller.cpp]: `%s` 应该是 `%1`

`sync this folder %s` => `sync folder %1`

Comment (sun20121818) [reply src/message-poller.cpp]: OK

File: src/message-poller.cpp [hunk 3]
```diff
@@ -187,6 +189,12 @@ void MessagePoller::processNotification(const SyncNotification& notification)
         case SYNC_ERROR_ID_FOLDER_PERM_DENIED:
             msg = tr("Update to file %1 is denied by folder permission setting.").arg(path);
             break;
+        case SYNC_ERROR_ID_PERM_NOT_SYNCABLE:
+            msg = tr("No permission to sync this folder %1.").arg(path);
```
Comment (lins05) [inline src/message-poller.cpp]: `sync this folder %1` =>  `sync folder %1`

Comment (sun20121818) [reply src/message-poller.cpp]: OK

**Pull request merged**